dandiapi/analytics/tasks/__init__.py

```diff
@@ -70,7 +69,7 @@ def process_s3_log_file_task(bucket: LogBucket, s3_log_key: str) -> None:
 
     # short circuit if the log file has already been processed. note that this doesn't guarantee
     # exactly once processing, that's what the unique constraint on ProcessedS3Log is for.
-    if ProcessedS3Log.objects.filter(name=Path(s3_log_key).name, embargoed=embargoed).exists():
+    if ProcessedS3Log.objects.filter(name=s3_log_key.split('/')[-1], embargoed=embargoed).exists():
```

Comment:
These aren't really URLs, more like `PurePath`s, no?

Reply:
It follows the structure of a `PurePosixPath`, but not necessarily the semantics (so using methods like [PurePath.as_uri](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_uri) would be wrong).

It might still make sense to change the type of all the variables holding S3 paths (to `PurePosixPath` or maybe [yarl](https://github.com/aio-libs/yarl)), but that's beyond the scope of this change. Most usages are already strings, so this makes them internally consistent and stops using inappropriate methods on them.